proto/p4/p4runtime.proto

```diff
@@ -382,10 +382,25 @@ message Uint128 {
 message SetForwardingPipelineConfigRequest {
   enum Action {
     UNSPECIFIED = 0;
+    // Verify that the target can implement the given config. Do not modify the
+    // forwarding state in the target. Returns error if config is not provided.
     VERIFY = 1;
+    // Verify that the target can implement the given config and save it. Do not
+    // modify the forwarding state in the target. Any subsequent read/write
+    // requests must refer to fields in the new config. Returns error if config
+    // is not provided.
     VERIFY_AND_SAVE = 2;
+    // Verify, save and implement the given config. Clear the forwarding state
+    // in the target. Returns error if config is not provided.
     VERIFY_AND_COMMIT = 3;
+    // Verify, save and implement the given config, or the last saved config if
+    // the config is not provided. Update the forwarding state in the target by
+    // replaying the write requests since the last config was saved.
     COMMIT = 4;
```

Comment:
I think it should be an error if `COMMIT` doesn't happen right after a `VERIFY_AND_SAVE` or if a config is sent with `COMMIT`.